Add diagnostic presets to pre-industrial+concentrations configuration

blimlim opened the pull request:
This pull request adds _Standard_ and _Detailed_ diagnostic request presets for each submodel to the pre-industrial+concentrations configurations. Switching between the presets can be controlled by changing the relevant symlink in each submodel's control directory. 

The changes to these presets compared to the previous default diagnostic requests are:

**Atmosphere**
_Detailed:_ same as previous requests, though bugged wetland fraction field fld_s08i248, CO2 flux variables fld_s03i326 and fld_s03i327 which don't work in concentration mode, and fld_s03i173 which is named incorrectly are removed. Soil temp averaged over tiles added in.
_Standard:_ Same as _Detailed_ with most variables on model theta levels removed and most daily tiled data removed.

**Ocean**
_Detailed:_ Same as previous requests, with all 3D monthly variables duplicated to yearly output. Variable renaming is removed, so that vanilla names are used. Variables that were requested previously requested but were not output by the model are not included. 
__Standard__: Same as detailed but with several 3D monthly fields removed (kept at yearly frequency), as described [here](https://forum.access-hive.org.au/t/standard-ocean-outputs-for-esm1-5/2265/11?u=spencerwong). 

Both ocean diag_tables are generated via the `make_diag_table` script, and hence produce output with one variable per file.

**Ice**
_Detailed:_ Same as existing historical configuration (which included daily output for several fields), though with the bugged fields [identified here](https://forum.access-hive.org.au/t/issue-with-cice-diagnostic-dvidtt-in-access-esm-the-cmip6-pre-industrial-archive/2249/11?u=spencerwong) removed.
_Standard:_ Same as _Detailed_, with daily output removed. 

Ice history namelist settings have been moved from the `cice_in.nml` to the separate `ice_history.nml` to enable swapping via symlinks.

blimlim commented:
One thing to note: The addition of the separate `ice_history.nml` namelist to control CICE's output requires the latest changes to payu from this [pull request](https://github.com/payu-org/payu/pull/483) in order to work. CICE will not find the required settings and will crash without these changes. 

I don't think these changes have made their way into the dev version of payu yet, and so the configuration might not work currently if you try to run with it.

anton-seaice commented on ice/diagnostic_requests/ice_history_detailed.nml:
```diff
@@ -0,0 +1,128 @@
+  , histfreq       = 'd','m','x','x','x'
+  , histfreq_n     = 1, 1, 1, 1, 1
+  , hist_avg       = .true.
+  , history_file   = 'iceh'
+  , history_format = 'nc'
```
We'll never change these two lines, so I think leave them in `cice_in.nml`

blimlim replied:
Ah that's good to know! I've moved these two back into `cice_in.nml`

anton-seaice reviewed (commented):
The cice stuff looks good. is `diagnostics_requests` the best name for it ? I think `diagnostics_profiles` or `history_profiles` is better?

minghangli-uni commented:
For the ocean `diag_table`, I am wondering why both yearly and monthly frequency fields are included at the same time? Was this decided at the release meeting? At the end of the day, both frequencies contribute to the same diagnostics, with yearly diagnostics essentially being an average of the monthly ones. Storing both can create redundancy, causing larger datasets and inefficiencies in data management. And maintaining both frequencies might cause confusion and require more computational resources.

blimlim commented:
> For the ocean diag_table, I am wondering why both yearly and monthly frequency fields are included at the same time?

That's a good point, and not something I have a great reason for. When setting up the `diag_table_standard`, the idea was to keep the regularly used 3D fields at monthly frequency, and move the less used ones to yearly. Given that some fields were being kept at yearly frequencies, I wasn't sure whether it was important to then do the same for the more popular ones too, in case it would be convenient for people doing yearly analysis. You're right that it is quite redundant though.

For the _diag_table_detailed_ where all 3D variables are saved at monthly frequency, I agree it's probably not useful to keep the yearly 3D variables at all.

@minghangli-uni do you think the following would make more sense:

_diag_table_detailed_: 
- As is but remove all yearly 3D fields

_diag_table_standard_: 
- Keep the regularly used 3D fields at monthly frequency
- At yearly frequency, keep only the less regularly used fields (i.e. those which are not in the monthly frequency list)

Let me know if this would sound a bit better, or any other ideas/suggestions would be great too!

minghangli-uni commented:
Sounds good to me! Thanks @blimlim

blimlim commented:
> I think diagnostics_profiles or history_profiles is better?
Yep I think those are easier to understand – I've renamed the directories to `diagnostic_profiles` but also happy with either of those suggestions.

blimlim commented:
@minghangli-uni, I've made those changes for the yearly ocean data, let me know if it's looking ok.

@anton-seaice, for some reason I don't understand, the `dumpfreq` reverted to `m`. I've set it back to `y` but not sure why github is reporting the original value as monthly.

MartinDix commented:
The standard monthly means include  `4 |THETA AFTER TIMESTEP`  on model levels. I don't think this is useful. 

We should save 
`1|    1 |    0 |  407 |PRESSURE AT RHO LEVELS AFTER TS     |`
which is necessary for calculating a vertical integral of the tracers

minghangli-uni reviewed (dismissed):
The ocean diagnostics look good to me. Thanks, @blimlim

blimlim commented:
Thanks!
I've removed `THETA AFTER TIMESTEP` from the standard profile, and also the daily `LEAF AREA INDEX (CASA-CNP GLAI)`.

I've added in `PRESSURE AT RHO LEVELS AFTER TS` to both the standard and detailed preset, using `IDOM=8`, corresponding to ` &DOMAIN NAME="DALLRH"`

The detailed preset already had a copy of this variable, though on the domain ` &DOMAIN NAME="DALLTH",`:

```
 &STREQ IMOD= 1, ISEC=0, ITEM=407, IDOM=2, ITIM=3, IUSE=2 / 
 &STREQ IMOD= 1, ISEC=0, ITEM=407, IDOM=8, ITIM=3, IUSE=2 /
```
@MartinDix, is it important to keep both of these, or is one of the domain selections incorrect?

MartinDix commented:
Only the version on the rho levels gets through to the output file so you can remove the theta one. The model seems to just silently ignore the one with theta levels.

aidanheerdegen reviewed (approved):
Looks great, thanks for all the work and constructive input from others.

blimlim commented:
Great! Thanks everyone for your help!
